### PR TITLE
Drop Microsoft.AspNetCore.Http.Features from net5 & net6

### PR DESCRIPTION
**Pull request by trejjam**
## Changes

The `Microsoft.AspNetCore.Http.Features` is [part of the net6](https://docs.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/microsoft-aspnetcore-http-features-package-split).

This PR removes the dependency on the not existing package `Microsoft.AspNetCore.Http.Features 6.0` for the net6 target.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj [hunk 1]
```diff
@@ -15,23 +15,26 @@
...
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Features" Version="$(MicrosoftAspNetCoreHttpFeaturesPkgVer)" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Features" Version="$(MicrosoftAspNetCoreHttpFeaturesPkgVer)" />
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:37]: mm.. This would mean in a Net5.0 app, we'll bring 2.1 version of this package?

Comment (trejjam) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:37]: I can specify versions more tightly to target frameworks if you like. The variable is not used anywhere else. Should I create new variables? Or can I inline dependency versions to this csproj? It is unlikely that the version will ever change.

I did not want to create `MicrosoftAspNetCoreHttpFeaturesPkgVer` explosion (new variable for nearly each target). Also, I did want the tiniest change possible.

Comment (trejjam) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:37]: Hello @cijothomas, can you point me in the right direction?

TLDR: Version bound in the csproj - it's a fading out dependency, or to the props?

**Comment by trejjam** [comment]
Hello @cijothomas, can you point me in the right direction?

TLDR: Version bound in the csproj - it's a fading out dependency, or to the props?

**Comment by trejjam** [comment]
Ping @cijothomas

File: src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj:37]: Do we need this for net5.0 target actually? Will it not come from <FrameworkReference Include="Microsoft.AspNetCore.App" /> ?

Comment (trejjam) [reply src/OpenTelemetry.Instrumentation.AspNetCore/OpenTelemetry.Instrumentation.AspNetCore.csproj]: Yes, you are right, didn't see it before. Also removed

**Comment by trejjam** [comment]
Ready for re-review

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@trejjam could you add an entry to changelog, as this is public facing change.

**Review by pellared — APPROVED** [review]
